togetherly/lib/views/widgets/new_chore_dialog.dart

```diff
@@ -0,0 +1,400 @@
+
+class NewChoreDialog extends StatefulWidget {
+  const NewChoreDialog({super.key, this.assignedChildId});
+  final int? assignedChildId;
+
+  @override
+  State<NewChoreDialog> createState() => _NewChoreDialogState();
+}
+
+class _NewChoreDialogState extends State<NewChoreDialog> {
+  final _formKey = GlobalKey<FormState>();
+  final _titleController = TextEditingController();
+  final _descriptionController = TextEditingController();
+  final _pointsController = TextEditingController();
+  DateTime _dueDate = DateTime.now();
+  String _title = '';
+  String _description = '';
+  int _points = 0;
+  bool _isBonus = false;
+  bool _isShared = false;
+  bool _loading = false;
+  final List<Map<String, dynamic>> _repeatingWeekdays = [
+    {
+      'day': 'S',
+      'index': 0,
+      'selected': false,
+    },
+    {
+      'day': 'M',
+      'index': 1,
+      'selected': false,
+    },
+    {
+      'day': 'T',
+      'index': 2,
+      'selected': false,
+    },
+    {
+      'day': 'W',
+      'index': 3,
+      'selected': false,
+    },
+    {
+      'day': 'Th',
+      'index': 4,
+      'selected': false,
+    },
+    {
+      'day': 'F',
+      'index': 5,
+      'selected': false,
+    },
+    {
+      'day': 'Sa',
+      'index': 6,
+      'selected': false,
+    },
+  ];
+  final List<Child> _assignedPeople = [];
+
+  List<Child> peopleList = const [
+    Child(
+      familyId: 0,
+      name: 'Emma',
+      icon: ProfileIcon.bear,
+      totalPoints: 45,
+    ),
+    Child(
+      familyId: 0,
+      name: 'Jacob',
+      icon: ProfileIcon.dog,
+      totalPoints: 80,
+    ),
+    Child(
+      familyId: 0,
+      name: 'Natalie',
+      icon: ProfileIcon.cat,
+      totalPoints: 65,
+    ),
+    Child(
+      familyId: 0,
+      name: 'Robert',
+      icon: ProfileIcon.giraffe,
+      totalPoints: 30,
+    ),
+  ];
+
+  void submitChore(BuildContext context, ChoreProvider provider) {
+    if (_formKey.currentState!.validate()) {
+      final newChore = Chore(
+        title: _title,
+        description: _description,
+        points: _points,
+        dueDate: _dueDate,
+        isShared: _isShared,
+        assignedChildId: 1,
+      );
+      setState(() => _loading = true);
+      provider.addChore(newChore);
+      setState(() => _loading = false);
+      Navigator.of(context).pop();
+    }
+  }
+
+  String? validatePoints(String value) {
+    if (value.isEmpty) {
+      setState(() => _points = 0);
+    } else {
+      final numericRegex = RegExp(r'^[0-9]+$');
+      if (numericRegex.hasMatch(value)) {
+        setState(() => _points = int.tryParse(value) ?? 0);
+      } else {
+        return 'Value can only contain numeric characters';
+      }
+    }
+    return null;
+  }
+
+  String prettyDate(String date) {
+    List<String> dateTimeArr = date.split(' ');
+    List<String> dateArr = dateTimeArr.elementAt(0).split('-');
+    return '${dateArr.elementAt(1)}/${dateArr.elementAt(2)}/${dateArr.elementAt(0)}';
+  }
```

Comment:
I saw this method elsewhere and left a comment on it. If you're using it in multiple places, it may be a good candidate to move into the `utility` folder.

Reply:
Good point.